src/syntax/parser.rs

```diff
@@ -1184,7 +1193,7 @@ fn validate_args(p: &mut Parser, m: Marker) {
     }
 }
 
-fn validate_destruct_pattern(p: &mut Parser, m: Marker) {
+fn validate_destruct_pattern(p: &mut Parser, m: Marker, forbid_expressions: bool) {
```

Comment:
why do we need `forbid_expressions`?

Reply:
This function is also used for validating patterns in let bindings, but we only want to allow `(a.at(0),)` as a pattern in assignments.